src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -754,6 +753,66 @@ class OPENASSETIO_CORE_EXPORT Manager final {
   void entityExists(const EntityReferences& entityReferences, const ContextConstPtr& context,
                     const ExistsSuccessCallback& successCallback,
                     const BatchElementErrorCallback& errorCallback);
+
+  /**
+   * Callback signature used for a successful entity trait set query.
+   */
+  using EntityTraitsSuccessCallback = std::function<void(std::size_t, trait::TraitSet)>;
+
+  /**
+   * Retrieve the @ref trait_set of one or more @ref entity "entities".
+   *
+   * For example, this may be used to validate that a user-provided
+   * entity reference is appropriate for an operation.
+   *
+   * The trait set returned (via callback) for each @ref
+   * entity_reference varies according to the @p entityTraitsAccess
+   * access mode.
+   *
+   * If @ref access.EntityTraitsAccess.kRead "kRead" is given, the
+   * response will be an exhaustive trait set for the entity. If an
+   * entity does not exist, then the error callback will be invoked
+   * using the @ref errors.BatchElementError.ErrorCode.kEntityResolutionError
+   * "kEntityResolutionError" code.
+   *
+   * If @ref access.EntityTraitsAccess.kWrite "kWrite" is given, the
+   * response will be the minimal trait set required to publish to the
+   * entity reference. If an entity is read-only, the error callback
+   * will be invoked using the @ref errors.BatchElementError.ErrorCode.kEntityAccessError
+   * "kEntityAccessError" code.
+   *
+   * The trait set will include all traits that an entity possesses, not
+   * just those with `resolve`able properties, so that they can be used
+   * to determine the kind of entity.
```

Comment:
[Consider] This seems irrelevent to the host, I don't know why they would presume otherwise.

Reply:
I think this is quite important, as a host must not assume that a trait here means the data is resolvable or publishable (I probably would). Be good to add a note about using `managementPolicy` to determine which of these traits the manager may be able to resolve/persist data for.

Maybe invert the phrasing? (The trait set will include all traits that an entity possesses, note that not all traits will be resolvable or persistable by the manager....)

Reply:
Whatever phrasing you think works is cool - just feel its important to state that as well as some of those traits not being resolvable by a manager, just because a trait is in `entityTraits` for `kWrite` it doesn't mean the manager is capable of storing that traits properties during publish. I dont think that is obvious to a new-comer without already knowing about management policy.  The docs should try to bi-directionally link so which ever you come across first exposes the whole concept.

Reply:
> there's just no properties to fill in so it's pointless.

Btw, Im explicitly talking about the fact that a trait _with_ properties may not be resolvable by any given manager, or that manager storing those properties during publish, _regardless_ of whether that trait is required to characterize the entity.

Reply:
Initial rewrite in b095cd05, then paired with Tom on further updates in 188db754 and 1dfd5f05